From b2ef09f6c4034dec03898df7f8024fc58794b055 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Tue, 19 Jul 2011 12:40:19 +1000
Subject: Don't free anything in PreInit, let UnInit take care of it.

The server calls UnInit, avoid double free's by letting the server free
everything.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>

diff --git a/src/xf86HyperPen.c b/src/xf86HyperPen.c
index b0e5ac9..52a6e4f 100644
--- src/xf86HyperPen.c
+++ src/xf86HyperPen.c
@@ -973,14 +973,9 @@ xf86HypInit(InputDriverPtr      drv,
         }
     }
 
-    return rc;
-
  SetupProc_fail:
-    if (priv)
-        free(priv);
-    if (pInfo)
-        free(pInfo);
     return rc;
+
 }
 
 _X_EXPORT InputDriverRec HYPERPEN = {
-- 
cgit v0.10.2

